core/dom/element.js

```diff
@@ -480,17 +480,72 @@ CKEDITOR.dom.element.clearMarkers = function( database, element, removeFromDatab
 		 * Retrieve the bounding rectangle of the current element, in pixels,
 		 * relative to the upper-left corner of the browser's client area.
 		 *
-		 * @returns {Object} The dimensions of the DOM element including
-		 * `left`, `top`, `right`, `bottom`, `width` and `height`.
-		 */
-		getClientRect: function() {
+		 * You can pass additional parameter if the function should give absolute element position which can be used for positioning elements
+		 * inside scrollable areas.
+		 *
+		 * E.g. you could use this function with {@link CKEDITOR.dom.window#getFrame editor's window frame} to
+		 * calculate the absolute rectangle of the visible area of the editor's viewport.
+		 * The retrieved absolute rectangle could be used to position elements like toolbars or notifications (elements outside an editor)
+		 * to always keep them inside an editor viewport independently from the scroll position.
+		 *
+		 * ```javascript
+		 * var frame = editor.window.getFrame();
+		 * frame.getClientRect( true );
+		 * ```
+		 *
+		 * @since 4.10.0
+		 * @param {Boolean} [isAbsolute=false] The function will retrieve absolute rectangle of the currect element i.e. including scroll position.
```

Comment:
Typo currect -> current. Actually I think we can skip this word all along, it won't degradate it's meaning, as this is a method in element context.

Reply:
> including scroll position

Does it really include the scroll? It is my udnerstanding that it actually skips the scroll, it does not care about scroll.

Say, you have an iframe with 20 000 px of height, and you have scrolltop offset, which is positioned on top of the container. If you scroll the inner frame by 18 000 px, and request for absolute position of a point in top left are of this window, it will return "a little" of pixels :D it **will not** include these 18 000.

![image](https://user-images.githubusercontent.com/5353898/38140758-7b299126-3435-11e8-8c00-59291d1fff5d.png)

We need to be very, very careful about the wording it needs to clear and understandable.

Instead I'd suggest complete reword, and say that it will return an offset of element relative to top left corner of a **viewport** (do I recall correctly that it returns to a topmost viewport and not the document?).

Reply:
> Does it really include the scroll? It is my udnerstanding that it actually skips the scroll, it does not care about scroll.

Probably the better wording would be "including the scroll position of the topmost frame"?

But yeah, the most readable version would be "position relative to the topmost viewport" or just simply "absolute position".